libraries/botbuilder/src/botFrameworkAdapter.ts

```diff
@@ -254,8 +270,15 @@ export class BotFrameworkAdapter extends BotAdapter implements IUserTokenProvide
     constructor(settings?: Partial<BotFrameworkAdapterSettings>) {
         super();
         this.settings = { appId: '', appPassword: '', ...settings };
-        this.credentials = new MicrosoftAppCredentials(this.settings.appId, this.settings.appPassword || '', this.settings.channelAuthTenant);
-        this.credentialsProvider = new SimpleCredentialProvider(this.credentials.appId, this.credentials.appPassword);
+        
+        // If settings.useCertificateAuthentication is null, undefined or false, use app + password authentication
+        if (!settings || !settings.useCertificateAuthentication) {
```

Comment:
Does this sound javascript-legit? I want the default behavior (MicrosoftAppCredentials) to occur when settings are not provided, when settings.useCertificateAuthentication is false, null or undefined.

Reply:
This looks good, but below is a little more simplified 😄 

```suggestion
        if (!this.settings.useCertificateAuthentication) {
```

Reply:
Eventually we'll be able to use optional chaining `settings?.useCertificateAuthentication` or nullish coalescing `let x = foo ?? bar();` via TypeScript 3.7, once we upgrade 😄